dotFrontendOnboarding.md

```diff
@@ -0,0 +1,206 @@
+This command runs tests in the CI environment.
+
+  ```sh
+  ../mvnw -pl :dotcms-core-web test
+  ```
+
+By utilizing these commands, you can effectively manage and ensure the quality of the codebase through thorough testing and linting.  
+
+## Advanced Configurations
+
+In this section, you'll find advanced configuration options to optimize your development workflow and customize your environment. These configurations include how to insert the dotCMS license, map the Tomcat root directory for working with JSPs without recompiling and rebuilding the backend, overwrite language files, and enable application features through environment variables and feature flags.
+
+### Setting the dotCMS License
+
+- To set and activate your dotCMS license key to unlock full functionality of the platform, you need to map the license file in the `docker-compose` configuration. Add the volume mapping for the license file in the `volumes` section of your `docker-compose` file as shown below:
+
+  ```yaml
+  volumes:
+    - cms-shared:/data/shared
+    # MOUNT THE FOLLOWING LICENSE FILE
+    - ./your-dotCMS-license.zip:/data/shared/assets/license.zip
+  ```  
+
+### Mapping the Tomcat Root Directory
+
+- Mapping the Tomcat root directory allows you to work with JSP files directly without the need to recompile, rebuild, and redeploy the backend each time you make changes. This setup significantly speeds up the development process. To achieve this, add a new `volume` in your `docker-compose` file as shown below:
+
+  ```yaml
+  volumes:
+    - cms-shared:/data/shared
+    # MOUNT THE FOLLOWING FOLDER TO THE ROOT OF THE TOMCAT
+    - <project-path>/core/dotCMS/src/main/webapp/html:/srv/dotserver/tomcat-9.0.85/webapps/ROOT/html
+  ```
+
+### Overwriting Language Files
+
+- To overwrite language files and customize the application's localization without modifying the original source files, you can map a new `volume` in your `docker-compose` file as follows:
+
+  ```yaml
+  volumes:
+    - cms-shared:/data/shared
+    # OVERWRITE LANGUAGE PROPERTIES FILES
+    - <project-path>/core/dotCMS/src/main/webapp/WEB-INF/messages:/srv/dotserver/tomcat-9.0.85/webapps/ROOT/WEB-INF/messages
+  ```
+
+  This configuration maps the local folder containing your customized language properties files to the appropriate directory within the Tomcat web server, allowing for easy updates and maintenance of language-specific content.
+
+### Environment variables and feature flags
+
+- You can enable and manage various features of the application using environment variables and feature flags. To do this, add a new environment variable in your `docker-compose` file as shown below:
+
+  ```yaml
+  services:
+    dotcms:
+      image: dotcms/dotcms:trunk_149bea9
+      environment:
+        ...
+        # ENABLE THE FOLLOWING FEATURES
+        DOT_FEATURE_FLAG_SEO_IMPROVEMENTS: true
+        DOT_FEATURE_FLAG_SEO_PAGE_TOOLS: true
+        DOT_FEATURE_FLAG_NEW_BINARY_FIELD: true        
```

Comment:
I think this should have the format of:

```
'DOT_FEATURE_FLAG_SEO_IMPROVEMENTS': true
'DOT_FEATURE_FLAG_SEO_PAGE_TOOLS': true
'DOT_FEATURE_FLAG_NEW_BINARY_FIELD': true   
```

Reply:
Thanks Jalison, however in this case I think it would not be necessary. Quotes are generally not needed for the names of environment variables, although they can be useful for values.